### PR TITLE
Add weaver + version info to weaved assembly

### PR DESCRIPTION
**Pull request by kamil-mrzyglod**
References #232

### CODE REVIEW

File: FodyIsolated/InnerWeaver.cs [hunk 1]
```diff
@@ -170,6 +174,44 @@ void ExecuteWeavers()
...
+        if (cancelRequested)
+        {
+            return;
+        }
+
+        Logger.LogDebug("  Adding weaving info");
+        var startNew = Stopwatch.StartNew();
+
+        var td = new TypeDefinition(null, "FodyWeavingResults", TypeAttributes.Class | TypeAttributes.NotPublic);
+        var attrCtor = ModuleDefinition.ImportReference(
```
Comment (SimonCropp) [inline FodyIsolated/InnerWeaver.cs]: wont this result in a runtime reference?

Comment (kamil-mrzyglod) [inline FodyIsolated/InnerWeaver.cs]: @SimonCropp TBH I've been trying to find relevant info about this and haven't found it yet. On the other hand I can change the approach from importing a type reference to creating an attribute manually, something like:
```
var td = new TypeDefinition(null, "FodyWeavingResults", TypeAttributes.Class | TypeAttributes.NotPublic);
var md = new MethodDefinition(".ctor",
            MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.SpecialName |
            MethodAttributes.RTSpecialName, ModuleDefinition.TypeSystem.Void);

md.Body.Instructions.Add(Instruction.Create(OpCodes.Ldarg_0));
md.Body.Instructions.Add(Instruction.Create(OpCodes.Call,
            ModuleDefinition.ImportReference(typeof(Attribute).GetConstructor(new Type[] {}))));
md.Body.Instructions.Add(Instruction.Create(OpCodes.Nop));
md.Body.Instructions.Add(Instruction.Create(OpCodes.Ldarg_0));
md.Body.Instructions.Add(Instruction.Create(OpCodes.Ldarg_1));
md.Body.Instructions.Add(Instruction.Create(OpCodes.Ret));

var attr =
            new CustomAttribute(md);
```

**Comment by kamil-mrzyglod** [comment]
@SimonCropp Ok now I re-implemented how an attribute is generated.

**Comment by kamil-mrzyglod** [comment]
Hi @SimonCropp, could you tell when you'd have time to review this PR once again? :)

File: FodyIsolated.Tests/WeavingInfoTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public void WeavedAssembly_ShouldContainWeavedInfo()
+    {
+        var moduleDefinition =
+            ModuleDefinition.ReadModule(TestContext.CurrentContext.TestDirectory + "\\DummyAssembly.dll");
+
+        var innerWeaver = new InnerWeaver
+        {
+            References = string.Empty,
+            Logger = new Mock<ILogger>().Object,
+            AssemblyFilePath = TestContext.CurrentContext.TestDirectory + "\\DummyAssembly.dll",
```
Comment (GeertvanHorrik) [inline FodyIsolated.Tests/WeavingInfoTests.cs]: Use Path.Combine instead of string concatenation

File: FodyIsolated.Tests/WeavingInfoTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            ModuleDefinition = moduleDefinition,
+            DefineConstants = new List<string> { "Debug", "Release" },
+            Weavers = new List<WeaverEntry> {
+                new WeaverEntry
+                {
+                    TypeName = "ModuleWeaver",
+                    AssemblyName = "Sample.Fody",
+                    AssemblyPath = TestContext.CurrentContext.TestDirectory + "\\Sample.Fody.dll"
```
Comment (GeertvanHorrik) [inline FodyIsolated.Tests/WeavingInfoTests.cs]: Use Path.Combine instead of string concatenation

File: FodyIsolated.Tests/WeavingInfoTests.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                }
+            }
+        };
+
+        innerWeaver.Execute();
+
+        moduleDefinition =
+            ModuleDefinition.ReadModule(TestContext.CurrentContext.TestDirectory + "\\DummyAssembly.dll");
```
Comment (GeertvanHorrik) [inline FodyIsolated.Tests/WeavingInfoTests.cs]: Use Path.Combine instead of string concatenation

**Comment by kamil-mrzyglod** [comment]
@GeertvanHorrik Sure, I guess I'll have time in the next few days.

**Comment by kamil-mrzyglod** [comment]
@GeertvanHorrik Screenshot from a weaved assembly:
image

Generated code for `FodyGeneratedCodeAttribute`:
```
// Decompiled with JetBrains decompiler
// Type: FodyGeneratedCodeAttribute
// Assembly: DummyAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 73EC27C5-2337-4F22-BEE7-03F5E364D08E
// Assembly location: C:\MINE\Fody\FodyIsolated.Tests\bin\Release\DummyAssembly.dll

using System;

internal class FodyGeneratedCodeAttribute : Attribute
{
  internal readonly string Version;

  public FodyGeneratedCodeAttribute(string version)
  {
    this.Version = version;
  }
}
```

IL:
```
// Type: FodyGeneratedCodeAttribute 
// Assembly: DummyAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 73EC27C5-2337-4F22-BEE7-03F5E364D08E
// Location: C:\MINE\Fody\FodyIsolated.Tests\bin\Release\DummyAssembly.dll
// Sequence point data from decompiler

.class private auto ansi 
  FodyGeneratedCodeAttribute
    extends [mscorlib]System.Attribute
{

  .field assembly initonly string Version

  .method public hidebysig specialname rtspecialname instance void 
    .ctor(
      string version
    ) cil managed 
  {
    .maxstack 8

    IL_0000: ldarg.0      // this
    IL_0001: call         instance void [mscorlib]System.Attribute::.ctor()
    IL_0006: nop          
    IL_0007: nop          

    // [15 5 - 15 27]
    IL_0008: ldarg.0      // this
    IL_0009: ldarg.1      // version
    IL_000a: stfld        string FodyGeneratedCodeAttribute::Version
    IL_000f: ret          

  } // end of method FodyGeneratedCodeAttribute::.ctor
} // end of class FodyGeneratedCodeAttribute
```

Generated code for `FodyWeavingResults` class:

```
// Decompiled with JetBrains decompiler
// Type: FodyWeavingResults
// Assembly: DummyAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 73EC27C5-2337-4F22-BEE7-03F5E364D08E
// Assembly location: C:\MINE\Fody\FodyIsolated.Tests\bin\Release\DummyAssembly.dll

[FodyGeneratedCode("1.30.0")]
internal class FodyWeavingResults
{
  internal const string SampleFody = "0.0.0.0";
}
```

IL:
```
// Type: FodyWeavingResults 
// Assembly: DummyAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 73EC27C5-2337-4F22-BEE7-03F5E364D08E
// Location: C:\MINE\Fody\FodyIsolated.Tests\bin\Release\DummyAssembly.dll
// Sequence point data from decompiler

.class private auto ansi 
  FodyWeavingResults
{
  .custom instance void FodyGeneratedCodeAttribute::.ctor(string) 
    = (01 00 06 31 2e 33 30 2e 30 00 00 ) // ...1.30.0..
    // string('1.30.0')

  .field assembly literal string SampleFody = "0.0.0.0"
} // end of class FodyWeavingResults
```

**Comment by GeertvanHorrik** [comment]
Looks good! @SimonCropp , can you merge this one?

**Pull request merged**